c3po/job/utils/metadata.py

```diff
@@ -0,0 +1,72 @@
+from c3po.db.dao import user, link, song, artist, genre
+from c3po.db.common.base import session_factory
+from datetime import datetime
+from music_metadata_extractor import SongData
+
+session = session_factory()
+
+class Metadata:
+    def __init__(self, url):
+        self.url = url
+        self.data = SongData(url)
+
+    def insert(self):
+        new_link = self._insert_link(self.url)
+        if(new_link):
+            new_song = self._insert_song(self.data.track)
+            new_link.song_id = new_song.id
+            for artist_data in self.data.artists:
+                new_artist = self._insert_artist(artist_data)
+                new_artist_song = artist.ArtistSong(new_artist, new_song)
+                session.add(new_artist_song)
+        session.commit()
+
+    def _insert_link(self, url):
+        query = list(session.query(link.Link).filter(link.Link.url == url))
```

Comment:
Nitpick: Doesn't SQLAlchemy have a way to query only one item? Because the link either exists or it doesn't. So we can get rid of the `query[0]` below.

Reply:
The query has a `one()` method apparently. I'll use that :)

Reply:
But this involves handling two errors: `NoResultFound` and `MultipleResultsFound`. Is that fine?